### ISSUE TITLE
Bug in CookiesStorage leads to path mismatch

### ISSUE THREAD

**Issue by avkuzenkov** [ktorio/ktor#624]
```
internal fun Cookie.matches(requestUrl: Url): Boolean {
    val domain = domain?.toLowerCase() ?: error("Domain field should have the default value")
    val path = with(path) {
        val current = path?.toLowerCase() ?: error("Path field should have the default value")
        if (current.endsWith("/")) current else "$path/"
}
...
```

If path not endsWith '/' then non-lowercased value will be assigned to path and it will not match with lowercased requestPath.

Suggested fix:
```
   val path = path?.toLowerCase()?.let {
      if (it.endsWith('/')) it else "$it/"
   } ?: error("Path field should have the default value")

```

**Comment by e5l** [comment]
Hi, @avkuzenkov. Thanks for the report.
It looks like a fixed problem, but I'll add the case to the test set and check it before the next minor release.

**Comment by e5l** [comment]
Fixed in `1.0.0-beta-1`
